### ISSUE TITLE
Problem with encode_printable_string()

### ISSUE THREAD

**Issue by nplrkn** [gabhijit/hampi#37]
Hello, congrats on v0.2.0!  I'm just trying it out and I'm seeing a problem with printable string coding.

To repro it, try adding this test to the end of `codecs/mod.rs`.  The test fails on the tip, but passes on an old branch I had lying around.

```
    use crate::aper::{decode::decode_printable_string, encode::encode_printable_string};

    #[test]
    fn printable_string() {
        let mut data = AperCodecData::new();
        let s1 = "hello".to_string();
        encode_printable_string(&mut data, Some(1), Some(150), true, &s1, false).unwrap();
        let s2 = decode_printable_string(&mut data, Some(1), Some(150), true).unwrap();
        assert_eq!(s1, s2);
    }
```

Failure message: 
```
thread 'aper::tests::printable_string' panicked at 'assertion failed: `(left == right)`
  left: `"hello"`,
 right: `"\u{0}\u{0}\u{0}\u{0}\u{0}"`', codecs/src/aper/mod.rs:465:9
```

**Comment by gabhijit** [comment]
I believe this issue occurs because the `data` parameter passed to the `decode` is not reset after encode. I believe in the least, the 'internal' codec state needs to be reset. I believe we need a method that produces a new `AperCodecData` from an existing one and then use that API and then try to decode. 
Perhaps this is not a bug per se, but may be adding that API in the `AperCodecData` would help?

**Comment by nplrkn** [comment]
Based on my first look the other day, it seemed to be deeper than that.  I'll dig into it later today and get back to you.

**Comment by gabhijit** [comment]
Merged #38 closing this now! Thanks for submitting the issue and the PR.
